service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportController.java

```diff
@@ -87,11 +93,11 @@
             DistributionEnvelopeDocument distributionEnvelope = extractDistributionEnvelope(reportElementsMap
                 .get(DISTRIBUTION_ENVELOPE));
             validate(distributionEnvelope);
-            POCDMT000002UK01ClinicalDocument1 clinicalDocument = extractClinicalDocument(distributionEnvelope);
 
-            validate(clinicalDocument);
+            List<POCDMT000002UK01ClinicalDocument1> clinicalDocuments = extractClinicalDocument(distributionEnvelope);
 
-            encounterReportService.transformAndPopulateToGP(clinicalDocument, messageId, headerValues);
+            validate(getDocumentWithMaxEffectiveDate(clinicalDocuments));
```

Comment:
`getDocumentWithMaxEffectiveDate(clinicalDocuments)` could be extracted to variable and used here and in the next line.